src/Arcus.WebApi.Tests.Unit/Security/Authorization/JwtTokenReaderTests.cs

```diff
@@ -77,5 +90,102 @@ public void Constructor_WithTokenValidationParametersAndDiscoveryEndpoint_Succee
             // Assert
             Assert.NotNull(jwtTokenReader);
         }
+
+        [Fact]
+        public void Constructor_WithClaimCheck_Succeeds()
+        {
+            // Arrange
+            var tokenValidationParameters = new TokenValidationParameters();
+            string openIdConnectDiscoveryUri = "https://login.microsoftonline.com/common/v2.0/.well-known/openid-configuration";
+
+            var claimCheck = new Dictionary<string, string>
+            {
+                {"aud", "368e1166-2235-48c7-8685-48d1fc24a9de"},
+                {"oid", "fa323e12-e4b8-4e22-bb2a-b18cb4b76301"},
+                {"aad", "ecc1e1c0-61bc-4e87-bd41-02333aa485bd"}
+            };
+
+            // Act
+            var jwtTokenReader = new JwtTokenReader(tokenValidationParameters, openIdConnectDiscoveryUri, claimCheck);
+
+            // Assert
+            Assert.NotNull(jwtTokenReader);
+        }
+
+        [Fact]
+        public async Task Constructor_ValidateClaimCheck_Succeeds()
+        {
+            // Arrange
+            var token = CreateJwt(Issuer, Authority, PrivateKey, DaysValid);
+            var handler = new JwtSecurityTokenHandler();
+            var securityToken = handler.ReadJwtToken(token);
+
+            var claimCheck = new Dictionary<string, string>
+            {
+                {"aud", Issuer},
+                {"oid", "fa323e12-e4b8-4e22-bb2a-b18cb4b76301"},
+                {"aad", "ecc1e1c0-61bc-4e87-bd41-02333aa485bd"}
+            };
+
+            // Act
+            var jwtTokenReader = new JwtTokenReader(claimCheck);
+            var isValid = jwtTokenReader.ValidateClaimCheck(securityToken);
+
+            // Assert
+            Assert.True(isValid);
+        }
+
+        [Fact]
+        public async Task Constructor_ValidateClaimCheck_Invalid()
+        {
+            // Arrange
+            var token = CreateJwt(Issuer, Authority, PrivateKey, DaysValid);
+            var handler = new JwtSecurityTokenHandler();
+            var securityToken = handler.ReadJwtToken(token);
+
+            var claimCheck = new Dictionary<string, string>
+            {
+                {"aud", Issuer},
+                {"oid", "fa323e12-e4b8-4e22-bb2a-b18cb4b76301"},
```

Comment:
These checks have the same ID's as the ones that are created for the identities. It should be good that these are both be generated instead of hard-coded like this.

Reply:
done